waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PollingFederationService.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2016-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.service.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import com.hotels.bdp.waggledance.api.model.AbstractMetaStore;
+import com.hotels.bdp.waggledance.api.model.MetaStoreStatus;
+import com.hotels.bdp.waggledance.core.federation.service.PopulateStatusFederationService;
+
+public class PollingFederationService {
+
+  private final static Logger log = LoggerFactory.getLogger(PollingFederationService.class);
+
+  private final PopulateStatusFederationService populateStatusFederationService;
+  private Map<String, MetaStoreStatus> previous = new HashMap<>();
+
+  public PollingFederationService(PopulateStatusFederationService populateStatusFederationService) {
+    this.populateStatusFederationService = populateStatusFederationService;
+  }
+
+  // status-polling-delay, default every 5 minutes
+  @Scheduled(fixedDelayString = "${status-polling-delay:300000}")
```

Comment:
OK, so there is a default here, not sure what the pros/cons are of having it here as opposed to in the 'WaggleDanceConfiguration'

Reply:
we'd have the default value in two places. If I remove it here you'd get a weird error if you set it to null. Doesn't matter much, I found this easiest